crates/common/certificate/src/lib.rs

```diff
@@ -135,48 +149,16 @@ impl KeyCertPair {
         key_kind: &KeyKind,
         not_before: OffsetDateTime,
     ) -> Result<CertificateParams, CertificateError> {
-        KeyCertPair::check_identifier(id, config.max_cn_size)?;
-        let mut distinguished_name = rcgen::DistinguishedName::new();
-        distinguished_name.push(rcgen::DnType::CommonName, id);
-        distinguished_name.push(rcgen::DnType::OrganizationName, &config.organization_name);
-        distinguished_name.push(
-            rcgen::DnType::OrganizationalUnitName,
-            &config.organizational_unit_name,
-        );
-
-        let mut params = CertificateParams::default();
-
-        params.distinguished_name = distinguished_name;
+        let mut params = Self::create_csr_parameters(config, id, key_kind)?;
 
         let not_after = not_before + Duration::days(config.validity_period_days.into());
         params.not_before = not_before;
         params.not_after = not_after;
 
-        params.is_ca = rcgen::IsCa::Ca(rcgen::BasicConstraints::Unconstrained); // IsCa::SelfSignedOnly is rejected by C8Y
```

Comment:
This is the change failing the system tests.

I forgot to inject this back to self-signed certificate creation.

Reply:
Fixed by https://github.com/thin-edge/thin-edge.io/pull/3268/commits/950ebcb48ab941dd2d37cd9f753b87ef0cf1946f